src/fides/api/ops/models/policy.py

```diff
@@ -269,7 +276,25 @@ def save(self, db: Session) -> FidesBase:
 
     @classmethod
     def create(cls, db: Session, *, data: Dict[str, Any]) -> FidesBase:  # type: ignore[override]
-        """Validate this object's data before deferring to the superclass on update"""
+        """Validate this object's data before deferring to the superclass on create"""
+        policy_id: Optional[str] = data.get("policy_id")
+
+        if not policy_id:
+            raise common_exceptions.RuleValidationError(
+                "Policy id must be specified on Rule create."
+            )
+
+        policy = Policy.get_by(db=db, field="id", value=policy_id)
+        if not policy:
+            raise common_exceptions.RuleValidationError(
+                "Policy id must be specified on Rule create."
+            )
+        existing_consent_rules = policy.get_rules_for_action(ActionType.consent)
+
+        if existing_consent_rules:
```

Comment:
i must be missing something obvious - but this seems like it's going to do this check on _any_ `Rule` that's being created? i would have thought we only wanted to run this check if the rule being added is a consent rule, i.e. if its `action_type == ActionType.consent`

Reply:
Thank you good catch, that was my intent here!